Ecocredit Activity page

mhagel opened the pull request:
## Description

Closes: regen-network/regen-registry#741

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

A page that shows a table of all ecocredit transactions

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

haveanicedavid reviewed (approved):
Looking good! Note - I added several suggestions, but most of the `sx` stuff isn't necessarily something you need to change. I wanted to show how using it would avoid `makeStyles` entirely in many cases. I prefer it because it avoids boilerplate, requires less overall code,  and keeps `makeStyles` to classes which affect multiple elements - but i'm fine either way

I'm pre-approving and wrote way more than I intended to here in this review, but do ask that we add a type to the response of `getTxns` in `/ecocredit.ts` before we merge

mhagel commented on web-components/src/components/table/useTablePagination.ts:
```diff
@@ -8,21 +8,25 @@ import { TablePagination, TablePaginationProps } from './TablePagination';
  */
 export function useTablePagination(
-  rowsPerPageOptions: number[] = [5],
+  rowsPerPageOptions: RowsPerPageOptions = { options: [5] },
```
I added an option `default` property. In my case, I had options of `[5, 10, 25, 50]`, but I needed the default to be `10` instead of the first option

mhagel commented on web-components/src/components/table/useTablePagination.ts:
```diff
@@ -8,21 +8,25 @@ import { TablePagination, TablePaginationProps } from './TablePagination';
   const [page, setPage] = useState<number>(0);
-  const [rowsPerPage, setRowsPerPage] = useState<number>(rowsPerPageOptions[0]);
+  const [rowsPerPage, setRowsPerPage] = useState<number>(
+    rowsPerPageOptions.default || 5,
```
there is a default key now, optional

mhagel commented on web-registry/src/lib/ecocredit.ts:
```diff
@@ -0,0 +1,106 @@
+
+export const getTxs = async (
+  paginationParams?: URLSearchParams,
+): Promise<any[]> => {
```
Thanks, I hate leaving `any`s unless totally necessary. In this case, it is a pretty burly object, but luckily I can pull it from regen-js!

mhagel replied:
I was able to pull the type from regen-js!

mhagel commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,252 @@
+  }, [fetchData, getReadableMessages, setCountTotal]);
+
+  return (
+    <StyledTableContainer className={styles.root}>
```
done

mhagel replied:
@haveanicedavid  I really like it! Feels a bit more maintainable. There is a performance tradeoff to be aware of: 
https://mui.com/system/basics/#performance-tradeoff
but otherwise I think it will be nice for development.

mhagel replied:
I left a comment, but I think github lost it...

I like this! There is a potential performance hit to be aware of: https://mui.com/system/basics/#performance-tradeoff

but otherwise it should be good for dev.

mhagel commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,252 @@
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody className={styles.tableBody}>
```
done

mhagel commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,252 @@
+                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                .map((tx: any) => {
+                  return (
+                    <StyledTableRow
+                      className={styles.noWrap}
+                      tabIndex={-1}
+                      key={tx.txhash}
+                    >
```
done

mhagel commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,252 @@
+                      tabIndex={-1}
+                      key={tx.txhash}
+                    >
+                      <StyledTableCell className={styles.grey}>
```
done

mhagel commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,252 @@
+                        </a>
+                      </StyledTableCell>
+                      <StyledTableCell>
+                        <Label className={styles.label}>{tx.messages}</Label>
```
this one requires modifying the Label component - it wasn't obvious how I should do this, so leaving for now.

mhagel replied:
This requires a modification of the Label component. I need to figure out the best way to do this, so leaving for now

mhagel commented on web-components/src/components/table/useTablePagination.ts:
```diff
@@ -53,8 +57,8 @@ export function useTablePagination(
 
-  const paginationProps = {
-    rowsPerPageOptions,
+  let paginationProps = {
```
back to const

mhagel commented on web-registry/src/components/molecules/Statistic.tsx:
```diff
@@ -0,0 +1,74 @@
+  return (
+    <div className={styles.root}>
+      <Label className={styles.label}>{label}</Label>
+      <div className={styles.data}>
```
done

mhagel commented on web-registry/src/components/molecules/Statistic.tsx:
```diff
@@ -0,0 +1,74 @@
+  };
+
+  return (
+    <div className={styles.root}>
```
done

mhagel commented on web-registry/src/components/molecules/Statistic.tsx:
```diff
@@ -0,0 +1,74 @@
+      <Label className={styles.label}>{label}</Label>
+      <div className={styles.data}>
+        {arrow && (
+          <div className={styles.circle}>
```
done

mhagel commented:
@haveanicedavid I talked to Erika and decided to make an `EcocreditsSection` component to handle the slightly different padding in these pages. Also, I addressed some of your comments so requesting another review.

blushi commented on web-registry/src/components/organisms/CreditActivityTable.tsx:
```diff
@@ -0,0 +1,247 @@
+const ROWS_PER_PAGE_OPTIONS = { options: [5, 10, 20, 50], default: 10 };
+
+const READABLE_NAMES: Dictionary<string> = {
+  '/regen.ecocredit.v1alpha1.MsgSend': 'send',
+  '/regen.ecocredit.v1alpha1.MsgRetire': 'retire',
+  '/regen.ecocredit.v1alpha1.MsgCreateClass': 'create credit class',
+  '/regen.ecocredit.v1alpha1.MsgCancel': 'cancel',
+  '/regen.ecocredit.v1alpha1.MsgCreateBatch': 'create batch',
```
the messages type names could eventually be pulled from regen-js directly (once MsgService impl is merged and released), could you add a TODO?

mhagel replied:
done
